simpleui-server/src/SuiZMQ.ts

```diff
@@ -0,0 +1,180 @@
+                    zmq_request_packet = `<request COMMAND="${cmd.cmd}" valueName="${cmd.valueName}"/>`;
+                    // log zmq details
+                    Logger.log( SuiData.requestNum <= 5 ? LogLevel.INFO : LogLevel.DEBUG,
+                        `zmq details:\n\ttimeout:\t${timeout}\n\tPort:\t\t${this.port}\n\tCMD:\t\t${JSON.stringify(cmd)}\n\tMsg:\t\t${zmq_request_packet}`);
+                }
+
+                // send request
+                this.socket.send(zmq_request_packet);
+            });
+
+
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `Could not create ZMQ socket at port ${this.port} got error: ${err}`);
+            process.exit(1);
+        }
+    }
+
+    connect(port: number) {
+        try {
+            Logger.log(LogLevel.VERBOSE, `ZMQ connecting to tcp://${this.hostname}:${port}`)
+            this.socket.connect(`tcp://${this.hostname}:${port}`)
+        } catch (e) {
+            Logger.log(LogLevel.ERROR, `Could not connect to tcp://${this.hostname}:${port} Got error: ${e}`);
+            process.exit(1);
+        }
+    }
+
+    send(msg: string) {
+        try{
+            this.socket.send(msg);
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `could not send zmq message:\n${msg} got error: ${err}`);
+        }
+    }
+
+    close() {
+        if (this.socket.closed === false) {
+            this.socket.close();
+        }
+    }
+
+    set_timeout(ms: number) {
+        try {
+            this.socket.connect_timeout = ms;
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `COULD NOT SET ZMQ TIMEOUT ${ms}`);
+        }
+    }
+}
+
+
+/**
+ * This class handles the creation of the zmq map
+ * and the destruction of the sockets
+ */
+export class zmq_wrapper {
+    socket_map: Map<number, ZMQ_Socket_Wrapper>;
+
+    constructor(port_list: Array<number>) {
+        this.socket_map = new Map();
+
+
+        port_list.forEach( (port) => {
+            const zmq_wrapper_instance = new ZMQ_Socket_Wrapper('svcmachineapps', port);
+            this.socket_map.set(port, zmq_wrapper_instance);
+        });
+
+        process.on('SIGINT', () => this.handleApplicationExit('SIGINT'));
+
+        process.on('SIGTERM', () => this.handleApplicationExit('SIGTERM'));
+
+        process.on('SIGWINCH', () => this.handleApplicationExit('SIGWINCH'));
+
+    }
+
+    get(port: number) {
+        return this.socket_map.get(port);
+    }
+
+    size(): number {
+        return this.socket_map.size;
+    }
+
+    handleApplicationExit(signalName: string) {
+        Logger.log(LogLevel.INFO, `ZMQ_Socket_Wrapper recieved signal ${signalName}`);
+        process.exit(1)
+        /*
+        this.socket_map.forEach((zmq_wrapper_instance, port) => {
+            Logger.log(LogLevel.INFO, `closing zmq port: ${port}`);
+            zmq_wrapper_instance.close();
+            // close event listeners (message, error, item_added)?
+        });'
+        */
```

Comment:
remove?

Reply:
I uncommented it in the ['sui_zmq fixes for jim'](https://github.com/greg-ses/simpleui/pull/44/commits/0e8fd71eb32c934083683c93a34b3592e6462716) commit. However, now I am debating if we should close the sockets at all. I dont think it would be bad if we just did `process.exit(1)` instead, assuming the `process.exit(1)` can 'overpower' the zmq sockets and shut them off

Reply:
So, ultimately when your program dies any resources are going to get freed, so it's not possible to have a traditional memory/resource leak if the OS just terminates your program and frees its memory space.

That said, closing your stuff on exit is good practice for several reasons:

1. There's no guarantee that whoever is calling your function(s) intends on shutting down the application, maybe they just needed to use your component for a little while and now they're done with it.
2. In the case of remote resources, they may get confused/sad if you just stop communicating with them without properly closing the connection. 
3. If anything needs to get written to disk or some other resource when your program exits it may get lost or corrupted if you just let the whole program die

Reply:
That makes sense. I don't want simpleui's zmq sockets to potentially mess up the c++ zmq sockets so I'll keep this as it is